File: integration-tests-java/src/main/java/io/micronaut/jimmer/java/it/config/TenantFilter.java
package io.micronaut.jimmer.java.it.config;

import io.micronaut.jimmer.java.it.entity.TenantAwareProps;
import jakarta.inject.Singleton;
import org.babyfish.jimmer.sql.filter.Filter;
import org.babyfish.jimmer.sql.filter.FilterArgs;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

@Singleton
public class TenantFilter implements Filter<TenantAwareProps> {

    private final Logger LOGGER = LoggerFactory.getLogger(this.getClass());

    @Override
    public void filter(FilterArgs<TenantAwareProps> args) {
        LOGGER.info("args: {}", args);
        LOGGER.info("args.getTable(): {}", args.getTable());
        args.where(args.getTable().tenant().eq("a"));
    }
}

File: integration-tests-kotlin/src/main/kotlin/io/micronaut/jimmer/kotlin/it/config/TenantFilter.kt
package io.micronaut.jimmer.kotlin.it.config

import io.micronaut.jimmer.kotlin.it.entity.TenantAware
import io.micronaut.jimmer.kotlin.it.entity.tenant
import jakarta.inject.Singleton
import org.babyfish.jimmer.sql.kt.ast.expression.eq
import org.babyfish.jimmer.sql.kt.filter.KFilter
import org.babyfish.jimmer.sql.kt.filter.KFilterArgs

@Singleton
class TenantFilter(
    private val tenantProvider: TenantProvider,
) : KFilter<TenantAware> {
    override fun filter(args: KFilterArgs<TenantAware>) {
        tenantProvider.tenant.let {
            args.apply {
                where(table.tenant.eq(it))
            }
        }
    }
}
